### DISCUSSION TITLE
Cannot find module '@ioc:Adonis' modules within ts-jest

### DISCUSSION THREAD

**Discussion by wrbwilkinson** [adonisjs/core#3120] [Help]
Hi

I've been struggling to figure out why, I'm getting this error only in jest (using `ts-jest`):

`Cannot find module '@ioc:Adonis/Core/Logger' from 'app/Exceptions/Handler.ts'`

If anyone has any ideas, don't hesitate to share them.

What I've done already:
- checked in debug logs if `ts-jest` using the correct `tsconfig` file (same as my Adonis project, which runs fine, no errors there)
- Tried using `adonis-jest` error still persist
- Reset `jest` cache

**Comment by smuralimohan** [comment]
@wrbwilkinson 
Could you find a solution for this? A small thing, but such a big hurdle.... Doh!

**Reply by wrbwilkinson** [reply]
Unfortunately I haven't found a solution, the only thing a found was AdonisJS uses a transformer after compilation that changes the file path and I believe it's where the issue is

**Reply by Julien-R44** [reply]
You guys should look at this : https://github.com/zakodium/adonis-jest

**Reply by smuralimohan** [reply]
Thanks @Julien-R44 for the suggestion. I tried using it, but to no avail. I raised an issue (https://github.com/zakodium/adonis-jest/issues/7 ) there also 2 weeks ago,  but no response yet.

**Reply by carlmathisen** [reply]
@smuralimohan See if you're able to isolate the error in a new repository with a blank Adonis project and your jest setup, and then publish the repo on GitHub. It will be much easier to help by looking at the entire project setup.
